lib/audited/auditor.rb

```diff
@@ -430,18 +438,21 @@ def require_comment
         alias_method "#{attr_name}_callback".to_sym, attr_name
       end
 
-      def auditing_enabled
-        run_conditional_check(audited_options[:if]) &&
-          run_conditional_check(audited_options[:unless], matching: false) &&
+      def auditing_enabled(action = nil)
+        run_conditional_check(audited_options[:if], action: action) &&
+          run_conditional_check(audited_options[:unless], action: action, matching: false) &&
           self.class.auditing_enabled
       end
 
-      def run_conditional_check(condition, matching: true)
+      def run_conditional_check(condition, action:, matching: true)
         return true if condition.blank?
-        return condition.call(self) == matching if condition.respond_to?(:call)
-        return send(condition) == matching if respond_to?(condition.to_sym, true)
+        result = case condition
+        when Proc then condition.call(self, action)
```

Comment:
I thought about passing `action` as Symbol but `Audit.action` is also a String so it's more consistent this way.